pkg/sql/alter_column_type_test.go

```diff
@@ -0,0 +1,521 @@
+	wg.Add(1)
+	go func() {
+		if _, err := sqlDB.Exec(`ALTER TABLE t.test ALTER COLUMN x TYPE STRING`); err != nil {
+			t.Error(err)
+		}
+		wg.Done()
+	}()
+
+	<-swapNotification
+
+	row := sqlDB.QueryRow("SHOW CREATE TABLE t.test")
+	var scanName, create string
+	if err := row.Scan(&scanName, &create); err != nil {
+		t.Error(err)
+	}
+	expected := `CREATE TABLE test (
+	x INT8 NULL,
+	FAMILY "primary" (x, rowid)
+)`
+	if create != expected {
+		t.Fatalf("expected %s, found %s", expected, create)
+	}
+
+	// Let the schema change process continue.
+	waitBeforeContinuing <- struct{}{}
+	wg.Wait()
+
+	row = sqlDB.QueryRow("SHOW CREATE TABLE t.test")
+	if err := row.Scan(&scanName, &create); err != nil {
+		t.Error(err)
+	}
+	expected = `CREATE TABLE test (
+	x STRING NULL,
+	FAMILY "primary" (x, rowid)
+)`
+	if create != expected {
+		t.Fatalf("expected %s, found %s", expected, create)
+	}
+}
+
+// TestAlterColumnTypeFailureRollback tests running a primary key
+// change on a table created in the same transaction.
+func TestAlterColumnTypeFailureRollback(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	ctx := context.Background()
+	params, _ := tests.CreateTestServerParams()
+	s, sqlDB, kvDB := serverutils.StartServer(t, params)
+	defer s.Stopper().Stop(ctx)
+
+	if _, err := sqlDB.Exec(
+		`SET enable_alter_column_type_general = true;`); err != nil {
+		t.Error(err)
+	}
+
+	_, err := sqlDB.Exec(`
+CREATE DATABASE t;
+CREATE TABLE t.test (x STRING);
+INSERT INTO t.test VALUES ('1'), ('2'), ('HELLO');
+ALTER TABLE 
+t.test ALTER COLUMN x TYPE INT;
+`)
+	actual := err.Error()
+	expected := "pq: could not parse \"HELLO\" as type int: strconv.ParseInt: parsing \"HELLO\": invalid syntax"
+
+	if actual != expected {
+		t.Fatalf("expected %s, got %s", expected, actual)
+	}
+
+	desc := sqlbase.GetTableDescriptor(kvDB, "t", "test")
+	// Expect to find two drop mutations, one to drop the Column, one to drop the
+	// Computed Column swap.
+	if len(desc.Mutations) != 2 {
+		t.Fatalf("expected to find 2 mutations, but found %d", len(desc.Mutations))
+
+		if desc.Mutations[0].Direction != sqlbase.DescriptorMutation_DROP {
+			t.Fatal("expected to be a drop mutation")
+			if desc.Mutations[0].GetColumn().Name != "x_1" {
+				t.Fatal("expected to be dropping column x_1")
+			}
+		}
+
+		if desc.Mutations[0].Direction != sqlbase.DescriptorMutation_DROP {
+			t.Fatal("expected to be a drop mutation")
+		}
+
+		column := desc.Mutations[0].GetColumn()
+		if column == nil {
+			t.Fatal("expected column to be non-nil")
+		}
+		if column.Name != "x_1" {
+			t.Fatalf("expected to column to have name x_1, found: %s", column.Name)
+		}
+
+		if desc.Mutations[1].Direction != sqlbase.DescriptorMutation_DROP {
+			t.Fatal("expected to be a drop mutation")
+		}
+
+		if desc.Mutations[1].GetComputedColumnSwap() == nil {
+			t.Fatal("expected to find a ComputedColumnSwap mutation with direction drop")
+		}
+
+	}
+}
+
+func TestAddDropColumnDuringAlterColumnType(t *testing.T) {
```

Comment:
I dont understand what this test is specifically testing / why dropping a column is problematic?